### PR TITLE
fix: Client lib checks whether the http response is empty

### LINKED ISSUE #590: [V2] Http client lib should check whether the http response is empty
Since we have core-command V2 API  which can execute Get command with **ds-returnevent** parameter, so the DS will return **empty content** when ds-returnevent set to **no**. 

And the empty content will cause json unmarshal error, so the Http client lib should check whether the http response is empty.

https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-command/2.x#/default/get_device_name__name___command_

### PR DESCRIPTION
**Pull request by weichou1229**
Close #590

Signed-off-by: weichou <weichou1229@gmail.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/.github/Contributing.md.

## What is the new behavior?
Since we have core-command V2 API which can execute Get command with ds-returnevent parameter, so the DS will return empty content when ds-returnevent set to no.
And the empty content will cause json unmarshal error, so the Http client lib should check whether the http response is empty.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

File: v2/clients/http/utils/request.go [hunk 1]
```diff
@@ -76,7 +80,7 @@ func PostRequestWithRawData(
 	return nil
 }
 
-// Helper method to make the put JSON request and return the body
+// PutRequest make the put JSON request and return the body
```
Comment (cloudxxx8) [inline v2/clients/http/utils/request.go:83]: makes

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]
LGTM

**Review by lenny-intel — DISMISSED** [review]
LGTM, once address @cloudxxx8 comment

File: v2/clients/http/utils/request.go (hunk 1 above)
Comment (weichou1229) [reply v2/clients/http/utils/request.go]: Modified.

**Review by lenny-intel — APPROVED** [review]
LGTM

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**